src/lib.rs

```diff
@@ -246,6 +246,25 @@ impl<T: Translation + Clone> Mapping<T> {
         }
         Ok(())
     }
+
+    /// Applies the provided function to a number of PTEs corresponding to a given memory range.
+    ///
+    /// The virtual address range passed to the callback function may be expanded compared to the
+    /// `range` parameter, due to alignment to block boundaries.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`MapError::RegionBackwards`] if the range is backwards.
+    ///
+    /// Returns [`MapError::AddressRange`] if the largest address in the `range` is greater than the
+    /// largest virtual address covered by the page table given its root level.
+    pub fn walk_range<F>(&self, range: &MemoryRegion, f: &mut F) -> Result<(), MapError>
+    where
+        F: FnMut(&MemoryRegion, &Descriptor, usize),
+    {
+        self.root.walk_range(range, f)?;
```

Comment:
Nit: maybe just return the result of `self.root.walk_range(range, f)`?

Reply:
Excellent point.